### PR TITLE
merge jitsi-utils-kotlin and jitsi-utils

### PR DESCRIPTION
**Pull request by bbaldino**

### CODE REVIEW

**Comment by bbaldino** [comment]
@ibauersachs wrote you an email about this, but not sure if it was the right address.

**Comment by bbaldino** [comment]
@ibauersachs I found that kotlin has an OSGi-ready version, so I used that instead...hopefully that should work fine then.

**Comment by ibauersachs** [comment]
The address is correct, thanks for the heads-up. I think the changes are okay and if something is missing I'll deal with a follow-up PR.
But is there a reason to keep the Maven submodule? I think it just complicates the build files.

**Comment by bbaldino** [comment]
> But is there a reason to keep the Maven submodule? I think it just complicates the build files.

We've been thinking lately about more, smaller artifacts in a repo rather than a single big one, so I left it thinking we may want to take advantage of that in the future.  Normally I'd say get rid of it and we can add it back, but in the past it's been such a pain to migrate to submodules my thought was to leave it.

File: jitsi-utils/pom.xml [hunk 1]
```diff
@@ -127,32 +120,90 @@
...
+                        <id>java-test-compile</id>
                         <phase>test-compile</phase>
                         <goals>
                             <goal>testCompile</goal>
                         </goals>
                     </execution>
                 </executions>
             </plugin>
+            <!-- This plugin is necessary for free/extensions functions defined in the kotlin code to be
+            visible to kotlin code using this lib.  See https://youtrack.jetbrains.com/issue/KT-9164. -->
```
Comment (ibauersachs) [inline jitsi-utils/pom.xml:186]: I assume you tried the other method mentioned in this issue?
```
<Include-Resource>
     /=src/main/resources/,
     META-INF=target/classes/META-INF
 </Include-Resource>
```

Comment (ibauersachs) [reply jitsi-utils/pom.xml:186]: (That would go into <instructions> below line 141)

**Comment by ibauersachs** [comment]
> > But is there a reason to keep the Maven submodule? I think it just complicates the build files.
> 
> We've been thinking lately about more, smaller artifacts in a repo rather than a single big one, so I left it thinking we may want to take advantage of that in the future. Normally I'd say get rid of it and we can add it back, but in the past it's been such a pain to migrate to submodules my thought was to leave it.

Keeping things in advance is not something I like too much, but if you think this will really in this repo, keep it.

File: jitsi-utils/pom.xml (hunk 1 above)
Comment (bbaldino) [reply jitsi-utils/pom.xml:186]: Cool, looks like this worked, I'll do it this way.

**Comment by bbaldino** [comment]
> > > But is there a reason to keep the Maven submodule? I think it just complicates the build files.
> > 
> > 
> > We've been thinking lately about more, smaller artifacts in a repo rather than a single big one, so I left it thinking we may want to take advantage of that in the future. Normally I'd say get rid of it and we can add it back, but in the past it's been such a pain to migrate to submodules my thought was to leave it.
> 
> Keeping things in advance is not something I like too much, but if you think this will really happen in this repo, keep it.

Ok, I ended up getting rid of it.

File: src/main/kotlin/org/jitsi/utils/logging2/Foo.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.utils.logging2
+
+class Foo {
```
Comment (ibauersachs) [inline src/main/kotlin/org/jitsi/utils/logging2/Foo.kt:19]: Uhh...?

File: pom.xml [hunk 3]
```diff
@@ -60,15 +199,6 @@
                     </compilerArgs>
                 </configuration>
             </plugin>
-            <plugin>
```
Comment (ibauersachs) [inline pom.xml:63]: Is this causing issues with Kotlin?

File: src/main/kotlin/org/jitsi/utils/logging2/Foo.kt (hunk 2 above)
Comment (bbaldino) [reply src/main/kotlin/org/jitsi/utils/logging2/Foo.kt]: Oops.

File: pom.xml (hunk 3 above)
Comment (bbaldino) [reply pom.xml:63]: Nope, this removal was unintentional.  Added it back.

File: pom.xml [hunk 4]
```diff
@@ -17,41 +17,180 @@
...
+                            META-INF=target/classes/META-INF
+                        </Include-Resource>
+                    </instructions>
+                </configuration>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.8.1</version>
+                <version>3.5.1</version>
```
Comment (bgrozev) [inline pom.xml:167]: Change from 3.8.1 to 3.5.1?

Comment (bbaldino) [reply pom.xml]: Nice catch, must have been this version in the submodule file. Fixed.

**Pull request merged**